backend/src/services/release.ts

```diff
@@ -330,50 +339,29 @@ function getSemverQueryBounds(querySemver: string) {
   const semverRangeStandardised = semver.validRange(querySemver, { includePrerelease: false })
 
   if (!semverRangeStandardised) {
-    throw BadReq(`Semver range, '${querySemver}' is invalid `)
+    throw BadReq(`Semver range, '${querySemver}' is invalid.`)
   }
 
   const [expressionA, expressionB] = semverRangeStandardised.split(' ')
 
-  let lowerSemver: string = '',
-    upperSemver: string = '',
-    lowerInclusivity = false,
-    upperInclusivity = false
+  let lowerInclusivity, lowerSemver, upperInclusivity, upperSemver
+
+  //If lower semver
   if (expressionA.includes('>')) {
-    if (expressionA.includes('=')) {
-      lowerSemver = expressionA.replace('>=', '')
-      lowerInclusivity = true
-    } else {
-      lowerSemver = expressionA.replace('>', '')
-    }
-    //do check for expressionB
-    if (expressionB) {
-      if (expressionB.includes('=')) {
-        upperSemver = expressionB.replace('<=', '')
-        upperInclusivity = true
-      } else {
-        upperSemver = expressionB.replace('<', '')
-      }
-    }
+    lowerInclusivity = expressionA.includes('>=')
+    lowerSemver = expressionA.replace(/[<>=]/g, '')
   } else {
-    if (expressionA.includes('=')) {
-      upperSemver = expressionA.replace('<=', '')
-      upperInclusivity = true
-    } else {
-      upperSemver = expressionA.replace('<', '')
-    }
+    //upper semver
+    upperInclusivity = expressionA.includes('<=')
+    upperSemver = expressionA.replace(/[<=]/g, '')
   }
 
-  let lowerSemverObj: { metadata?: string | undefined; major: number; minor: number; patch: number } = {
-      major: NaN,
-      minor: NaN,
-      patch: NaN,
-    },
-    upperSemverObj: { metadata?: string | undefined; major: number; minor: number; patch: number } = {
-      major: NaN,
-      minor: NaN,
-      patch: NaN,
-    }
+  if (expressionB) {
+    upperInclusivity = expressionB.includes('<=')
+    upperSemver = expressionB.replace(/[<=]/g, '')
+  }
+
+  let lowerSemverObj, upperSemverObj
```

Comment:
There should both be typed. Also defining multiple values in a single line is something we should probably avoid.

Reply:
Yeah removed the typing because I wanted it to be undefined, forgetting that I can give the possible type of undefined!